pymongo/pool.py

```diff
@@ -550,7 +550,8 @@ def unpin(self):
             self.close_socket(ConnectionClosedReason.STALE)
 
     def hello_cmd(self):
-        if self.opts.server_api or self.hello_ok:
+        if self.opts.server_api or self.hello_ok or self.opts.load_balanced:
+            self.op_msg_enabled = True
```

Comment:
I think it's worth adding a comment here saying why. Something like "Handshake spec requires us to use OP_MSG+hello command for the initial handshake in load balanced or versioned api mode."